src/Server/TextDocument.php

```diff
@@ -79,7 +57,7 @@ public function documentSymbol(TextDocumentIdentifier $textDocument): array
      */
     public function didOpen(TextDocumentItem $textDocument)
     {
-        $this->updateAst($textDocument->uri, $textDocument->text);
+        $this->project->getDocument($textDocument->uri)->updateAst($textDocument->text);
```

Comment:
What do you think is "large" in that context? But I think I am talking BS here because the parser itself (for a single document) will always be synchronous, if we don't spawn a seperate process (which would probably be much more overhead especially on Windows).